gridstatus/isone_api/isone_api.py

```diff
@@ -195,8 +195,8 @@ def _handle_demand(
     @support_date_range("DAY_START")
     def get_realtime_hourly_demand(
         self,
-        date,
-        end_date: str | None = None,
+        date: str | pd.Timestamp = "latest",
+        end: str | pd.Timestamp | None = None,
```

Comment:
Can we add a `verbose` parameter to these methods to be used for logging like in our other `get_` methods?

Reply:
yes, but I'll state that I'm skeptical of the addition of arguments that are only touched by decorators and not in the function itself. If there is some precedent/resource supporting this approach in python I would love to read it. IMO those should be parameters of the decorator rather than the function. These will likely fail static type checking and have to be manually ignored.

Reply:
`verbose` isn't used by the `support_date_range` decorator. It should be used within the function itself to show information about the URLs being retrieved. Take a look for example at any of the `get_` methods in `isone.py`.

Reply:
ah got it. Yeah to me that's what logging levels should be used for.

Reply:
I agree there's a better way, but I'm not sure what it looks like with logging levels. If I'm in the interpreter running this function, how would I turn on/off logging if we used logging levels? Right now, I do:

```
# Logging
ISONEAPI().get_realtime_hourly_demand(date, end, verbose=True)
# No logging
ISONEAPI().get_realtime_hourly_demand(date, end, verbose=False)
```

Reply:
You import the logger and change the level in the interpreter/REPL/script
```
from gridstatus.gs_logging import logger
```

Example `INFO`
```
logger.setLevel(logging.INFO)
df = api.get_realtime_hourly_demand(date="2024-10-06", end="2024-10-08", locations=[".Z.MAINE", ".Z.NEWHAMPSHIRE"])
```
![CleanShot 2024-10-14 at 18 09 27@2x](https://github.com/user-attachments/assets/0bded319-a8e8-4a0a-a824-be584af0ed14)

Example `DEBUG`
```
logger.setLevel(logging.DEBUG)
df = api.get_realtime_hourly_demand(date="2024-10-06", end="2024-10-08", locations=[".Z.MAINE", ".Z.NEWHAMPSHIRE"])
```
![CleanShot 2024-10-14 at 18 11 01@2x](https://github.com/user-attachments/assets/ce074ce7-e025-4579-a588-c2b888769799)

Reply:
I pushed a commit to make the logger a bit more dynamic in setting a level at a script level.

I'm still getting double output on certain log calls, can't quite figure out why that's occurring only some of the time, but it is likely a less than ideal import somewhere I haven't tracked down yet, but will once I get to the logging ticket.

![CleanShot 2024-10-14 at 18 14 43@2x](https://github.com/user-attachments/assets/8fab7335-0660-464f-a649-5394deef29d4)

Reply:
Going to remove those example logs from the method. It appears that the DEBUG setting already outputs the URL.

Reply:
Thanks, this makes sense.